File: connectors/connector-msp/src/main/java/com/taskadapter/connector/msp/MSPFactory.java
package com.taskadapter.connector.msp;

import com.google.gson.JsonElement;
import com.google.gson.JsonParseException;
import com.taskadapter.connector.common.ConfigUtils;
import com.taskadapter.connector.definition.Descriptor;
import com.taskadapter.connector.definition.FileSetup;
import com.taskadapter.connector.definition.PluginFactory;
import com.taskadapter.model.Field;

import java.util.List;

public class MSPFactory implements PluginFactory<MSPConfig, FileSetup> {
    private static final Descriptor DESCRIPTOR = new Descriptor(MSPConnector.ID, "Microsoft Project");

    @Override
    public List<Field<?>> getAllFields() {
        return MspField.fields;
    }

    @Override
    public List<Field<?>> getDefaultFieldsForNewConfig() {
        return MspField.fields;
    }

    @Override
    public MSPConnector createConnector(MSPConfig config, FileSetup setup) {
        return new MSPConnector(setup);
    }

    @Override
    public Descriptor getDescriptor() {
        return DESCRIPTOR;
    }

    @Override
    public JsonElement writeConfig(MSPConfig config) {
        return ConfigUtils.createDefaultGson().toJsonTree(config);
    }

    @Override
    public MSPConfig readConfig(JsonElement config) throws JsonParseException {
        return ConfigUtils.createDefaultGson().fromJson(config, MSPConfig.class);
    }

    @Override
    public MSPConfig createDefaultConfig() {
        return new MSPConfig();
    }
}

File: connectors/connector-msp/src/main/scala/com/taskadapter/connector/msp/MspFactory.scala
package com.taskadapter.connector.msp

import com.google.gson.{JsonElement, JsonParseException}
import com.taskadapter.connector.common.ConfigUtils
import com.taskadapter.connector.definition.{Descriptor, FileSetup, PluginFactory}
import com.taskadapter.model.Field

class MSPFactory extends PluginFactory[MSPConfig, FileSetup] {
  private val DESCRIPTOR = Descriptor(MSPConnector.ID, MSPConfig.DEFAULT_LABEL)

  override def getAllFields: java.util.List[Field[_]] = MspField.fields

  override def createConnector(config: MSPConfig, setup: FileSetup) = new MSPConnector(setup)

  override def getDescriptor = DESCRIPTOR

  override def writeConfig(config: MSPConfig): JsonElement = ConfigUtils.createDefaultGson.toJsonTree(config)

  @throws[JsonParseException]
  override def readConfig(config: JsonElement): MSPConfig = ConfigUtils.createDefaultGson.fromJson(config, classOf[MSPConfig])

  override def createDefaultConfig = new MSPConfig

  override def getDefaultFieldsForNewConfig: java.util.List[Field[_]] = MspField.fields
}

